test/jest-light-runner/src/index.js

```diff
@@ -0,0 +1,56 @@
+import { Piscina } from "piscina";
+import supportsColor from "supports-color";
+import { MessageChannel } from "worker_threads";
+import os from "os";
+
+/** @typedef {import("@jest/test-result").Test} Test */
+
+const piscina = new Piscina({
+  filename: new URL("./worker-runner.js", import.meta.url).href,
+  maxThreads: os.cpus().length / 2,
+  env: {
+    // Workers don't have a tty; we whant them to inherit
+    // the color support level from the main thread.
+    FORCE_COLOR: supportsColor.stdout.level,
+    ...process.env,
+  },
+});
+
+export default class LightRunner {
+  #config;
+
+  constructor(config) {
+    this.#config = config;
+  }
+
+  /**
+   * @param {Array<Test>} tests
+   * @param {*} watcher
+   * @param {*} onStart
+   * @param {*} onResult
+   * @param {*} onFailure
+   */
+  runTests(tests, watcher, onStart, onResult, onFailure) {
+    return Promise.all(
```

Comment:
Should it be `Promise.allSettled`? If `onFailure` returns a rejected promise, other pending tests should not end.

Reply:
`onFailure` returns a resolved promise when it's called to report a failure. If it rejects, there is a bug in the test runner (not in a test) so we can exit and report the bug.